src/test/java/educationalproject/programmingstuff/tests_integrations/EndpointsProcessingTests.java

```diff
@@ -0,0 +1,171 @@
+
+    @Autowired
+    private UserService userService;
+
+    private RequestBuilder request;
+
+    private List<User> userEntities;
+
+    private List<UserResponseDto> userResponseDtos;
+
+    private String expectedResponse;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeEach
+    private void prepareTestData() {
+
+        List<User> users= TestDataCreator.getTestUsers();
+        userRepository.saveAllAndFlush(users);
+
+    }
+
+    @Test
+    @Transactional
+    void givenGetEndpointUsersWithNoParam_whenEndpointTriggered_thenReturnAllUsers() throws Exception {
+
+        //Given
+        request = MockMvcRequestBuilders.get("/users");
+        userEntities = userRepository.findAll();
+        userResponseDtos = userResponseMapper.makeUsersResponseOf(userEntities);
+        expectedResponse = objectMapper.writeValueAsString(userResponseDtos);
+
+        //When
+        ResultActions resultActions = mockMvc.perform(request);
+
+        //Then
+        resultActions
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedResponse));
+    }
+
+    @Test
+    @Transactional
+    void givenGetEndpointUsersWithName_whenEndpointTriggered_thenReturnUsersWithThisName() throws Exception {
+
+        //Given
+        String testName = "John";
+        request = MockMvcRequestBuilders.get("/users").param("name", testName);
+        userEntities = userRepository.getUsersByName(testName);
+        userResponseDtos = userResponseMapper.makeUsersResponseOf(userEntities);
+        expectedResponse = objectMapper.writeValueAsString(userResponseDtos);
+
+        //When
+        ResultActions resultActions = mockMvc.perform(request);
+
+        //Then
+        resultActions
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedResponse));
+    }
+
+    @Test
+    @Transactional
+    void givenPostEndpointUsersWithNewUser_whenEndpointTriggered_thenReturnThatUserWithNewId() throws Exception {
+
+        //Given
+
+        String testName = "Sasha";
+        String testSurname = "Grey";
+
+        RequestBuilder request = prepareRequest(testName, testSurname);
+
+        int usersTotalBeforeNewOneCreation = getAmountOfRegisteredUsers();
+
+        //When
+        MvcResult mvcResult = mockMvc.perform(request)
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //Then
+        MockHttpServletResponse response = mvcResult.getResponse();
+        String content = response.getContentAsString();
+        UserResponseDto userResponseDto = objectMapper.readValue(content, UserResponseDto.class);
+
+        assertNotNull(userResponseDto.getId());
+        assertEquals(usersTotalBeforeNewOneCreation + 1, getAmountOfRegisteredUsers());
+        assertEquals(testName,userResponseDto.getUserName());
+        assertEquals(testSurname,userResponseDto.getSurname());
+        assertNull(userResponseDto.getOrders());
```

Comment:
Too complicated. You could do like previous tests, create an expected dto and compare with result. Or compare jsons.